kalite/testing/behave_helpers.py

```diff
@@ -45,6 +48,26 @@
 
 # Maximum time to wait when trying to find elements
 MAX_WAIT_TIME = 1
+# Maximum time to wait for a page to load.
+MAX_PAGE_LOAD_TIME = 3
+
+
+def click_and_wait_for_page_load(context, elem, wait_time=MAX_PAGE_LOAD_TIME):
+    """ Click an element and then wait for the page to load. Does this by
+    first getting an element on the page, clicking, and then waiting for the
+    reference to become stale. If the element doesn't become stale then it throws
+    a TimeoutException. (So if you pass an element to click that doesn't cause a 
+    page load, then you'll probably get a TimeoutException.)
+    context: a behave context
+    elem: a WebElement to click.
+    wait_time: Optional. Max wait time for the page to load. Has a default value.
+    """
+    # The body element should always be on the page.
+    wait_elem = context.browser.find_element_by_tag_name("body")
+    elem.click()
+    WebDriverWait(context.browser, wait_time).until(
```

Comment:
Yep, this seems like the right way to do it. (reference for others interested: http://www.obeythetestinggoat.com/how-to-get-selenium-to-wait-for-page-load-after-a-click.html )